pontoon/pretranslation/tasks.py

```diff
@@ -107,44 +113,49 @@ def pretranslate(self, project_pk, locales=None, entities=None):
                     string=string,
                     user=user,
                     approved=False,
+                    fuzzy=True,
                     active=True,
                     plural_form=plural_form,
                 )
 
-                translations.append(t)
-
                 index += 1
+                translations.append(t)
+                locale_set.add(locale)
 
                 if locale_resource not in tr_dict:
-                    tr_dict[locale_resource] = [index, 0]
+                    tr_dict[locale_resource] = index
+
                     # Add query for fetching respective TranslatedResource.
                     tr_filter.append(
                         Q(locale__id=locale.id) & Q(resource__id=entity.resource.id)
                     )
 
-                if locale.code not in locale_dict:
-                    locale_dict[locale.code] = [locale, index, 0]
-
                 # Update the latest translation index
-                tr_dict[locale_resource][0] = index
-                locale_dict[locale.code][1] = index
+                tr_dict[locale_resource] = index
 
-                # Increment number of translations (used to adjust stats)
-                tr_dict[locale_resource][1] += 1
-                locale_dict[locale.code][2] += 1
         log.info("Fetching pretranslations for locale {} done".format(locale.code))
 
     if len(translations) == 0:
         return
 
     translations = Translation.objects.bulk_create(translations)
 
-    # Update latest activity and unreviewed count for the project.
-    project.latest_translation = translations[-1]
-    project.unreviewed_strings += len(translations)
-    project.save(update_fields=["latest_translation", "unreviewed_strings"])
+    translation_pks = {translation.pk for translation in translations}
```

Comment:
A comment above this line would be nice, e.g. `# Run checks on all translations`.